lib/drivers/medtronicDriver.js

```diff
@@ -1004,42 +1009,46 @@ module.exports = function (config) {
         // sort by time
         postrecords = _.sortBy(postrecords, function(d) { return d.time; });
 
-        for (var j = 0; j < postrecords.length; ++j) {
-          var datum = postrecords[j];
-          switch (datum.type) {
-            case 'basal':
-              simulator.basal(datum);
-              break;
-            case 'bolus':
-              simulator.bolus(datum);
-              break;
-            case 'wizard':
-              simulator.wizard(datum);
-              break;
-            case 'smbg':
-              simulator.smbg(datum);
-              break;
-            case 'pumpSettings':
-              simulator.pumpSettings(datum);
-              break;
-            case 'deviceEvent':
-              if (datum.subType === 'status') {
-                simulator.suspendResume(datum);
-              } else if (datum.subType === 'alarm') {
-                simulator.alarm(datum);
-              } else if (datum.subType === 'prime') {
-                simulator.prime(datum);
-              } else if (datum.subType === 'reservoirChange') {
-                simulator.rewind(datum);
-              } else if (datum.subType === 'timeChange') {
-                simulator.changeDeviceTime(datum);
-              }
-              break;
-            default:
-              debug('[Hand-off to simulator] Unhandled type!', datum.type);
+        try {
```

Comment:
Why is this part in a `try` block too? All I can think of is if `postrecords` is empty, but that should be fine instead of the `for` loop... 🤔 Perhaps because of errors thrown in the simulator, I guess?